conanfile.py

```diff
@@ -32,25 +32,20 @@ def set_version(self):
         self.version = re.search("set\(CMAKE_PROJECT_VERSION (.+)\)", content).group(1).strip()
 
     def configure(self):
-        # If we're building static then build the world as static, otherwise
-        # stuff will probably break.
-        # If you want your shared build to also build the world as shared, you
-        # can invoke conan with -o "&:shared=False" -o "*:shared=False",
-        # possibly with --build=missing or --build=cascade as desired,
-        # but this is probably not necessary.
-        if not self.options.shared:
-            self.options["*"].shared = False
+        # The shared-ness of grpc and protobuf needs to match our shared-ness or stuff will break
+        self.options["grpc"].shared = self.options.shared
+        self.options["protobuf"].shared = self.options.shared
 
     def requirements(self):
-        self.requires('boost/[>=1.74.0]', transitive_headers=True)
+        self.requires('boost/[>=1.74.0]', transitive_headers=True, transitive_libs=True)
 
         # The SDK supports older grpc and protobuf, but these are the oldest
         # maintained conan packages.
-        self.requires('grpc/[>=1.48.4]', transitive_headers=True)
-        self.requires('protobuf/[>=3.17.1]', transitive_headers=True)
+        self.requires('grpc/[>=1.48.4]', transitive_headers=True, transitive_libs=True)
+        self.requires('protobuf/[>=3.17.1]', transitive_headers=True, transitive_libs=True)
 
         self.requires('xtensor/[>=0.24.3]', transitive_headers=True)
-        self.requires('abseil/[>=20230125.3]')
+        self.requires('abseil/[>=20230125.3]', transitive_libs=True)
```

Comment:
Hmm in fact I might be able to remove it from `xtensor` because it is a `header-library`, see https://docs.conan.io/2/reference/conanfile/methods/requirements.html#package-type-trait-inferring

Let me see if `abseil` needs to be there too or not, I added it because comparing the conan-generated build files to the ones generated from our CMake install, ours were pulling in all abseil into the link line

Reply:
just for my own understanding, why do you set some properties like `transitive_libs` or  `transitive_headers` for some libraries and not others? and how do you know which ones to set?

Reply:
That actually took me a while to figure out and I had to ask on github, but there's a bit more in the page I linked above outside of the anchor point: https://docs.conan.io/2/reference/conanfile/methods/requirements.html

A `requires` in and of itself just means that you need this thing to build our project, and the `transitive_meow` properties then determine whether those requirements are visible downstream. With `boost` for example the SDK uses Boost, but moreover there are places where Boost is part of our public API and appears in public header files, which means that a consumer of the SDK must also have the include paths for the Boost libraries to work properly.

Transitive libs is similar but for linking libraries instead of include paths to headers, I'm still trying to get my head around it 100% though